aten/src/ATen/native/sparse/cuda/SparseCUDATensorMath.cu

```diff
@@ -895,6 +895,7 @@ Tensor& _bmm_out_sparse_cuda(Tensor& result, const SparseTensor& self, const Ten
   size_t workspace_buffer_size = 0;
   void* workspace_buffer = nullptr;
 
+  deterministic |= globalContext().deterministic();
```

Comment:
`deterministic = deterministic || globalContext().deterministic()` is slightly better because you get to short circuit if `deterministic` is true already.